sound/soc/sof/loader.c

```diff
@@ -714,6 +714,7 @@ int snd_sof_load_firmware_raw(struct snd_sof_dev *sdev)
 	ret = request_firmware(&plat_data->fw, fw_filename, sdev->dev);
 
 	if (ret < 0) {
+		plat_data->fw = NULL;
```

Comment:
Are you sure you need this?
the sdev->pdata is devm_kzalloced during module loading. Furthermore if we are here the plat_data->fw must have been NULL.

Reply:
and on request_firmware() failure the plat_data->fw is set to NULL.

Reply:
I was too lazy and paranoid to trust `request_firmware` but now I see it's even documented.